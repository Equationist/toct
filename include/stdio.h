/* stdio.h - Standard Input/Output Library for CompilerKit C Compiler */

#ifndef _STDIO_H
#define _STDIO_H

/* Standard I/O types */
typedef struct {
    int _fd;
    int _flags;
    char* _buffer;
} FILE;

/* Standard streams */
extern FILE* stdin;
extern FILE* stdout; 
extern FILE* stderr;

/* Standard I/O functions */
int printf(const char* format, ...);
int fprintf(FILE* stream, const char* format, ...);
int sprintf(char* str, const char* format, ...);
int snprintf(char* str, int size, const char* format, ...);

int scanf(const char* format, ...);
int fscanf(FILE* stream, const char* format, ...);
int sscanf(const char* str, const char* format, ...);

int putchar(int c);
int puts(const char* s);
int fputs(const char* s, FILE* stream);
int fputc(int c, FILE* stream);

int getchar(void);
char* gets(char* s);
char* fgets(char* s, int size, FILE* stream);
int fgetc(FILE* stream);
int ungetc(int c, FILE* stream);

FILE* fopen(const char* filename, const char* mode);
FILE* freopen(const char* filename, const char* mode, FILE* stream);
int fclose(FILE* stream);
int fflush(FILE* stream);

long ftell(FILE* stream);
int fseek(FILE* stream, long offset, int whence);
void rewind(FILE* stream);
int feof(FILE* stream);
int ferror(FILE* stream);
void clearerr(FILE* stream);

/* File position constants */
#define SEEK_SET 0
#define SEEK_CUR 1
#define SEEK_END 2

/* End of file constant */
#define EOF (-1)

/* Buffer size */
#define BUFSIZ 8192

/* Null pointer */
#ifndef NULL
#define NULL ((void*)0)
#endif

#endif /* _STDIO_H */