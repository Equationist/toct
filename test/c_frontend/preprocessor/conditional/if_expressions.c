/* Test #if with expressions */
#define VERSION_MAJOR 2
#define VERSION_MINOR 1
#define DEBUG_LEVEL 3

#include <stdio.h>

int main() {
#if VERSION_MAJOR > 1
    printf("Version is 2.x or higher\n");
#else
    printf("Version is 1.x\n");
#endif

#if VERSION_MAJOR == 2 && VERSION_MINOR >= 1
    printf("Version 2.1 or higher detected\n");
#endif

#if DEBUG_LEVEL >= 3
    printf("Debug mode: VERBOSE\n");
#elif DEBUG_LEVEL >= 2
    printf("Debug mode: NORMAL\n");
#elif DEBUG_LEVEL >= 1
    printf("Debug mode: MINIMAL\n");
#else
    printf("Debug mode: OFF\n");
#endif

#if defined(VERSION_MAJOR) && !defined(VERSION_PATCH)
    printf("Major version defined, patch version not defined\n");
#endif

    return 0;
}