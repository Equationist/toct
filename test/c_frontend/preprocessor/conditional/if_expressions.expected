Version is 2.x or higher
Version 2.1 or higher detected
Debug mode: VERBOSE
Major version defined, patch version not defined